app/models/hmcts_common_platform/offence.rb

```diff
@@ -0,0 +1,75 @@
+module HmctsCommonPlatform
+  class Offence
+    attr_reader :data
+
+    def initialize(data)
+      @data = data
+    end
+
+    def id
+      data[:id]
+    end
+
+    def offence_code
+      data[:offenceCode]
+    end
+
+    def order_index
+      data[:orderIndex]
+    end
+
+    def offence_title
+      data[:offenceTitle]
+    end
+
+    def mode_of_trial
+      data[:modeOfTrial]
+    end
+
+    def start_date
+      data[:startDate]
+    end
+
+    def wording
+      data[:wording]
+    end
+
+    def allocation_decision_mot_reason_code
+      data.dig(:allocationDecision, :motReasonCode)
+    end
+
+    def laa_appln_reference_status_code
+      data.dig(:laaApplnReference, :statusCode)
+    end
+
+    def laa_appln_reference_status_date
+      data.dig(:laaApplnReference, :statusDate)
+    end
+
+    def laa_appln_reference_end_date
+      data.dig(:laaApplnReference, :effectiveEndDate)
+    end
+
+    def laa_appln_reference_status_description
+      data.dig(:laaApplnReference, :statusDescription)
+    end
+
+    def laa_appln_reference_laa_contract_number
+      data.dig(:laaApplnReference, :laaContractNumber)
+    end
+
+    def results
+      data[:judicialResults]&.map do |judicial_result_data|
```

Comment:
`Array(data[:judicialResults]).map...`

Reply:
This then returns an empty array if there are no judicialResults which is not the output we want and would then need to be changed to nil at another point